tests/kernels/test_kernels.py

```diff
@@ -14,68 +14,111 @@
 """
 Unit tests for the :mod:`pennylane` kernels module.
 """
-import pennylane as qml
-import pennylane.kernels as kern
+# pylint: disable=import-outside-toplevel
+import math
+import sys
+from functools import partial
+
 import pytest
 import numpy as np
+import pennylane as qml
+import pennylane.kernels as kern
 from pennylane import numpy as pnp
-import math
-import sys
 
 
-@pytest.fixture()
-def cvxpy_support():
+@pytest.fixture(name="cvxpy_support")
+def fixture_cvxpy_support():
+    """Fixture to determine whether cvxpy and cvxopt are installed."""
+    # pylint: disable=unused-import
     try:
         import cvxpy
         import cvxopt
 
         cvxpy_support = True
-    except:
+    except ModuleNotFoundError:
         cvxpy_support = False
 
     return cvxpy_support
 
 
 @pytest.fixture()
 def skip_if_no_cvxpy_support(cvxpy_support):
+    """Fixture to skip a test if cvxpy or cvxopt are not installed."""
     if not cvxpy_support:
         pytest.skip("Skipped, no cvxpy support")
 
 
-def _mock_kernel(x1, x2, history):
+def _mock_kernel(x1, x2, history, batch=None):
     """A kernel that memorizes its calls and encodes a fixed values for equal/unequal
     datapoint pairs."""
     history.append((x1, x2))
 
+    mult = 1.0 if batch is None else np.linspace(1, 5, batch)
+
     if x1 == x2:
-        return 1
-    else:
-        return 0.2
+        return 1 * mult
+    return 0.2 * mult
 
 
-def _laplace_kernel(x1, x2):
+def _laplace_kernel(x1, x2, batch=None):
     """The laplace kernel on scalar data."""
-    return np.exp(-math.fabs(x1 - x2))
+    mult = 1.0 if batch is None else np.linspace(1, 5, batch)
+
+    return mult * np.exp(-math.fabs(x1 - x2))
+
+
+def _diffable_kernel(x1, x2):
+    """A differentiable kernel"""
+    return qml.math.exp(-((x1 - x2) ** 2))
+
+
+def _jacobian_of_diffable_kernel(x1, x2):
+    """Analytic Jacobian of diffable_kernel"""
+    return -2 * (x1 - x2) * qml.math.exp(-((x1 - x2) ** 2))
 
 
 class TestKernelMatrix:
     """Tests kernel matrix computations."""
 
-    def test_simple_kernel(self):
+    @pytest.mark.parametrize("batch", [None, 1, 4])
+    def test_simple_kernel(self, batch):
         """Test square_kernel_matrix and kernel_matrix of the _mock_kernel above."""
         X1 = [0.1, 0.4]
         X2 = [0.1, 0.3, 0.5]
 
         K1_expected = pnp.array([[1, 0.2], [0.2, 1]])
         K2_expected = pnp.array([[1, 0.2, 0.2], [0.2, 0.2, 0.2]])
+        if batch is not None:
+            K1_expected = pnp.tensordot(np.linspace(1, 5, batch), K1_expected, axes=0)
+            K2_expected = pnp.tensordot(np.linspace(1, 5, batch), K2_expected, axes=0)
 
-        K1 = kern.square_kernel_matrix(X1, lambda x1, x2: _mock_kernel(x1, x2, []))
-        K2 = kern.kernel_matrix(X1, X2, lambda x1, x2: _mock_kernel(x1, x2, []))
+        K1 = kern.square_kernel_matrix(X1, partial(_mock_kernel, history=[], batch=batch))
```

Comment:
Just curious: is there any significant benefit to using `partial` over lambda functions?

Reply:
Good question. I tend to move away from lambdas. Here they probably would be fine to keep.
I think for me personally `partial` is more readable, but I'm not sure why that is the case, actually :D

Reply:
The answers [here](https://stackoverflow.com/questions/11828410/differences-between-functools-partial-and-a-similar-lambda) seem to suggest that there is a slight preference for `partial` because it can be a notch faster (not too important) and it maintains a connection to the original function, e.g. the docstring, which `lambda`s do not. In test suites I guess there is no major difference.